### PR TITLE
raidboss: Fix tts issues re: #3136

### LINKED ISSUE #3136: E12S Double Aero Finder still has tts
is the purpose of this `tts: null` to make it silent?

This still get tts.

https://github.com/quisquous/cactbot/blob/961a1ba7321f96d5d475b449b75b016a3d0c8b8e/ui/raidboss/data/05-shb/raid/e12s.js#L1607-L1616

I guess the reason is here, we didn't distinguish `null` and `undefined`

https://github.com/quisquous/cactbot/blob/ed708582d715cf304e354b2631f9766b175f12ae/ui/raidboss/popup-text.ts#L1183-L1190

### PR DESCRIPTION
**Pull request by valarnin**
Fixes #3136

### CODE REVIEW

**Comment by valarnin** [comment]
The comment below this change indicates that `false` should disable as well, but doesn't mention `''`. Should I update both the `else if` clause and the comment to include both?

File: ui/raidboss/popup-text.ts [hunk 1]
```diff
@@ -1184,7 +1184,9 @@ export class PopupText {
       let result = undefined;
       if (triggerHelper.triggerOptions.TTSText) {
         result = triggerHelper.valueOrFunction(triggerHelper.triggerOptions.TTSText);
-      } else if (triggerHelper.trigger.tts) {
+      } else if (triggerHelper.trigger.tts ||
+          triggerHelper.trigger.tts === null ||
+          triggerHelper.trigger.tts === '') {
```
Comment (quisquous) [inline ui/raidboss/popup-text.ts:1189]:
```suggestion
      } else if (triggerHelper.trigger.tts !== undefined) {
        // Allow null or false in this branch.
```

What about this to allow for null, false, and empty string?

Comment (valarnin) [reply ui/raidboss/popup-text.ts:1189]: Would also allow `NaN` and `0`, if both of those are fine, then sure?

Comment (quisquous) [reply ui/raidboss/popup-text.ts:1189]: I can go either way.  I think personally returning either of those here is sort of "undefined behavior" and it's fine if they do weird things.  If you think it's better to be explicit and ignore it, that seems fine too?

Comment (valarnin) [reply ui/raidboss/popup-text.ts]: Updated, with some more clear comments. Sorry for the double commit, I forgot to save the comments before I staged/commited

File: ui/raidboss/popup-text.ts [hunk 2]
```diff
@@ -1184,15 +1184,16 @@ export class PopupText {
...
+        // Allow null/false/NaN/0/'' in this branch.
         result = triggerHelper.valueOrFunction(triggerHelper.trigger.tts);
       } else if (triggerHelper.response) {
         const resp: ResponseField<RaidbossData> = triggerHelper.response;
         if (resp.tts)
           result = triggerHelper.valueOrFunction(resp.tts);
       }
 
-      // Allow false or null to disable tts entirely
+      // Allow falsey values to disable tts entirely
```
Comment (quisquous) [inline ui/raidboss/popup-text.ts:1196]: Thanks for the extra comment.

**Pull request merged**